### PR TITLE
Adding support for Mongo persistence

### PR DESCRIPTION
**Pull request by vtrifonov**
Hi again Jordi :) I've added support for persistence in MongoDB as this may ease the support for deployment and maintenance if you want to have a bigger state. I guess the MongoDB may even be reused for storing the configurations. Please let me know what you think about the implementation!

### CODE REVIEW

**Comment by jmartin82** [comment]
Hey I'm pretty sure of that is the way to go, but let me put on hold this PR for a few days. I'm working with the definitions file because I want simplify it (I don't want add more features and configs with a clear plan about it). When I finish I'would like one code review from you side. I'll keep you on touch.

**Comment by vtrifonov** [comment]
Sounds good! Just let me know when you're ready!

**Comment by jmartin82** [comment]
Hi,  can you cherry pick the commit related with the multiple methods and also the nested config folders into a new pull request? Maybe I can merge it faster.

**Comment by vtrifonov** [comment]
Sure, I'll prepare another merge request.

**Comment by vtrifonov** [comment]
Can you check now whether the changes are OK

File: mmock.go [hunk 1]
```diff
@@ -75,8 +75,14 @@ func getRouter(mocks []definition.Mock, dUpdates chan []definition.Mock) *route.
 }
 
 func loadVarsProcessorEngines(persistPath string) *persist.PersistEngineBag {
-	filePersist := persist.FilePersister{PersistPath: persistPath}
-	persistBag := persist.GetNewPersistEngineBag(filePersist)
+	var persister persist.EntityPersister
+	if strings.Index(persistPath, "mongodb://") < 0 {
```
Comment (jmartin82) [inline mmock.go]: It's ok now, but I want change it in a near future. We can add both engines (or more ) inside the EngineBag and let the user choose which want in the mock configuration through the engine field (already available) What do you think about it? Does it make sense for you? (I'm not sure about if it worth)

**Pull request merged**